app/controllers/pets_controller.rb

```diff
@@ -0,0 +1,53 @@
+class PetsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @pets = Pet.all
+  end
+
+  def new
+    # pet ペット情報を保存　petForm 画像を保存で分ける
+    @pet = Pet.new
+    @pet_images = PetForm.new
+
+  end
+
+  def create
+    @pet = current_user.pets.build pet_params
+    ActiveRecord::Base.transaction do
+      @pet.save!
+      @pet_imagaes = PetForm.new(pet_id: @pet.reload.id, pet_images: pet_images)
+      @pet_imagaes.save!
+    end
+
+    flash[:notice] = "登録完了しました。"
+    redirect_to pet_path @pet
+
+  rescue => e
+    if  @pet.errors.full_messages.present?
+      flash[:alert] = @pet.errors.full_messages
+    elsif  @pet_imagaes.errors.full_messages.present?
+      flash[:alert] = @pet_imagaes.errors.full_messages
+    end
+
+    redirect_to new_pet_path
+
+  end
```

Comment:
`e.record` でエラーが発生したリソースが取得できるので↓のような形でもいけます。

```suggestion
  rescue => e
    flash[:alert] = e.record.errors.full_messages
    redirect_to new_pet_path
  end
```